app/scripts/directives/oscFileInput.js

```diff
@@ -68,8 +69,12 @@ angular.module('openshiftConsole')
           });
         }
 
+        $('.clear-btn').on('click', function() {
```

Comment:
1. you should scope the `$('.clear-btn')` to only look within the range of this directive's DOM elements, if there were more than one of these on the page you wouldn't want them to affect each other.  To scope it you just pass the element as the second parameter `$('.clear-btn', element)`
2. we are trying to get into the habit of scoping calls to jquery on/off.  This lets you have multiple click handlers on the same node that can be independently turned off.  To scope a click handler it looks something like this `$('.clear-btn', element).on('click.fileInputClear')`   notice the `click.fileInputClear`  the first part before the `.` is the event, the part after is whatever you want to call your scope
3. If you add a jquery event handler in a directive you should clean it up within the `scope.$on('$destroy'` section, so add this `$('.clear-btn', element).off('click.fileInputClear')`